hazelcast/src/main/java/com/hazelcast/spi/impl/CountingMigrationAwareService.java

```diff
@@ -141,4 +145,12 @@ public boolean shouldOffload() {
         return migrationAwareService instanceof OffloadedReplicationPreparation
                 && ((OffloadedReplicationPreparation) migrationAwareService).shouldOffload();
     }
+
+    @Override
+    public ChunkSupplier newChunkSupplier(PartitionReplicationEvent event, Collection<ServiceNamespace> namespace) {
+        if (!(migrationAwareService instanceof ChunkedMigrationAwareService)) {
+            return null;
+        }
```

Comment:
what if we rather call the default method?
```java
if (!(migrationAwareService instanceof ChunkedMigrationAwareService)) {
        return ChunkedMigrationAwareService.super.newChunkSupplier(event, namespace);
}
```

Reply:
it seems safer to return null here since `CountingMigrationAwareService` delegates `FragmentedMigrationAwareService` . All `FragmentedMigrationAwareService` objects cannot be an implementation of `ChunkedMigrationAwareService`